pkg/build/api/types.go

```diff
@@ -185,6 +185,11 @@ type CustomBuildStrategy struct {
 	// a validation error is returned.
 	Image string `json:"image"`
 
+	// DockerRegistryAuthRef is used to provide Docker Registry authentication
+	// secret that is used for communication with authentication enabled Docker
+	// Registry. This have to reference existing Secret object.
+	DockerRegistrySecretRef *kapi.ObjectReference `json:"dockerRegistrySecretRef,omitempty"`
```

Comment:
Even just Secret is fine, it doesn't have to contain "Ref" in the name, we don't do that in other places that we use ObjectReference.

I'd go w/ RegistrySecret

Is this field going to exist in all 3 strategies?  if so i'm wondering if it should be factored out to a higher level.